Vic2ToHoI4/Source/HOI4World/HoI4World.cpp

```diff
@@ -551,6 +554,47 @@ bool HoI4::World::dominionIsReleasable(const Country& dominion, const Country& o
 }
 
 
+void HoI4::World::transferPuppetsToDominions()
+{
+	for (auto& country: countries | std::views::values )
+	{
+		std::map<std::string, std::set<std::string>> regionalPuppets; // <region, puppets>
+		for (const auto& puppetTag: country->getPuppets())
+		{
+			const auto& puppetItr = countries.find(puppetTag);
+			if (puppetItr == countries.end())
+			{
+				continue;
+			}
+			const auto& puppet = puppetItr->second;
+			if (puppet->isGeneratedDominion())
+			{
+				continue;
+			}
+			const auto& capital = puppet->getCapitalProvince();
+			if (!capital)
+			{
+				continue;
+			}
+			const auto& region = theRegions->getRegion(*capital);
+			if (region)
+			{
+				regionalPuppets[*region].insert(puppetTag);
+			}
+		}
+
+		for (const auto& [region, puppets]: regionalPuppets)
+		{
+			auto dominion = countries.find(*country->getDominionTag(region));
```

Comment:
You'll need to put country->getDominionTag(region) into a variable and explicitly handle the case where it returns nullopt.